### DISCUSSION TITLE
License question concerning temperaments

### DISCUSSION THREAD

**Discussion by polaronde** [GrandOrgue/grandorgue#1442] [Developers] [answered]
Hello everybody,

I am interested in opinions about license and usage rules, concerning a certain GO source file.

The reason is that I am implementing temperaments functionality for my organ apps and am looking for a set of factory temperaments to be included in an app.

GrandOrgue, specifically the file src/core/temperaments/GOTemperamentList.cpp, is a nice reference for temperaments data.

This file is under GPL which does not allow to use it in non-open source software.

Now I do not intend to use the source file but only the temperaments data (name, cent values), which by itself I think is not copyrighted.

Still that would mean to make use of the work of somebody else. Some GO developer(s) spent a considerable amount of time with searching and compiling the data of 103 temperaments.

Is it allowed to extract this data and use it in other software (stating GO as the source)? Or do I have to google for each of these temperaments and take the data from elsewhere (and keep notes of the process to be able to prove that I did not copy data of the GPLed file)? Or could the developer(s) who collected this data give permission to use their work in other (partly free, partly commercial) software?

Best regards,
Markus

**Comment by larspalo** [comment, marked as answer]
If I remember correctly I did most of the research work on the temperaments, and that information in itself is of course available also from other sources - so the cent deviations in themselves cannot be thought of as copyrighted.

If you look at the old mailing lists on Sourceforge you have this post:

https://sourceforge.net/p/ourorgan/mailman/message/29415405/

Which has an attachment:

https://sourceforge.net/p/ourorgan/mailman/attachment/1339854605.22029.YahooMailNeo%40web132106.mail.ird.yahoo.com/1/

That file contain the work material before inclusion in GO. That text file doesn't have any copyright header at all. I don't remember exactly how much could have been adjusted since that post, that's for you to research and compare.

However, I'll give you this hint to help you when you do your reserach: When we implemented temperaments a distinct choice was made to reference from the pitch of the tone for "a" so that the reference pitch of a1 (for instance 440 Hz) would not change when different temperaments was chosen (selected). Most of the academic literature on temperaments always reference from the tone "c" instead. The major part of my work contribution (except to collect the data to begin with) was to "transpose" the cent offsets from the litterature into our choice of having the reference pitch always staying the same. Thus, you should always find the third number from the end to be a "0" value. If you read and compare to other sources you'll see that that very note will, more or less always, have another value - but the relationships between all the tones will be the same.

**Reply by polaronde** [reply]
Many thanks for your response, Lars.

So I assume that it is okay to make use of the data that is in the old Sourceforge file, possibly with some corrections. I will happily refer to you and the GrandOrgue project as the creator / curator of the material and add links to GO and to that temperaments file.

In a first implementation I used "c" as the reference note, but not anymore, because from a discussion with a user I learned that for musicians it is fine to have "a" as the note whose frequency is not changed by the temperament. This does not exclude to let users define temperaments where "a" is detuned, as well.

**Reply by larspalo** [reply]
@polaronde Right, most musicians doesn't want the reference "a" to change, but it's of course possible to re-tune that note as well. Feel free to use the data in the text file. As I remember, it should be the same data as in GO source, but check for yourself to make sure.

**Comment by midimusic** [comment]
I too am working on temperaments and learning a lot as I go. Is it correct that the smallest pitch difference that can be detected by the human ear is 5 cents? This appears to be widely accepted.

Having looked pitch lists of the multitude of GO temperaments I see that nearly all of them are listed to two decimal places of a cent and in a great many of them most of the pitch differences are less than 5 cents.

Is it really necessary to have so many different scales when around 90% of them cannot be distinguished one from the other by listening to them?

I have no fixed opinion on this matter and would welcome any comments from musicians. Especially those who use and understand these various temperaments.

csw900

**Reply by larspalo** [reply]
@midimusic It's not that easy really, the correct answer is that it depends on the context. As a harmony (two notes as an interval) it's perfectly possible to discern down to 1 cent or even less (I can easily do this when I'm tuning any kind of interval, but mostly fifths and octaves). As a melody (a single line of notes played after each other) we are not at all that sensitive and a difference of 5 cents of a note in between two others most likely won't be noticed.

Actually, we had a test during a choir workshop when I did my church music studies. There a triad was played broken, and the middle tone (the major third) was gradually raised from perfectly pure to almost as high as a perfect fourth and it's really amazing how the human ear adapts the sound to what it wants to hear (the major triad in this case). But if you hear the triad played as a chord, the ear can be extremely sensitive indeed.

So, temperaments shouldn't really be judged as organized "scales" (diatonic or chromatic), scales are a completely different topic, if we talk music theory, than temperament is. Temperaments is all about how to adjust the intervals for harmonic reasons - not so much for melodic reasons, that is totally secondary even if it's true that the melody can get a different flavour in different temperaments.

To be honest, all of the historic temperaments were, at the time they were conceived, tuned by ear alone, listening for the number of beats (or lack thereof) of different intervals according to different rules. No digital tuners with .01 cent precision were ever used back then. And still today, the real (they who do it as a real handicraft) piano tuners tune by ear and counting the beats - except for setting the reference tone of course. The "cheaters" who use digital tuners would never get through the tests to be accepted in the piano tuners guild...

**Reply by polaronde** [reply]
Another interesting topic is dynamic tuning, in particular when it is applied not only to static chords but to the progressing music. When you hear a note, your brain remembers the heard frequency for a certain time. When you play another note within this time, it is favourable to have the new note be tuned to a nice ratio to the remembered frequency. After that time, this does not matter anymore. A dynamic tuning scheme that adapts to the played music can respect this.

If the played notes are known beforehand, then the dynamic tuning can even be applied prospectively i.e. such that the current notes and the future notes have to be minimally retuned. One might say that the algorithm does not only use the past, but also the future. And even if the subsequent notes are not known, an algorithm could try to predict them.

This is a fascinating subject, from a musical as well as from a mathematical perspective. At some point I would like to implement a dynamic tuning function myself. But there are so many other things to do...

Material to read: "Playing Music in Just Intonation - A Dynamically Adapting Tuning Scheme", https://arxiv.org/abs/1706.04338

**Comment by midimusic** [comment]
So not one of the hundreds of available scales is musically perfect in all respects - thus the normally used even tempered scale is perfect enough and is the most practical for everyday use. It amazes me that a piano tuner can tune this scale by listening for beats as most of its intervals must be purposly tuned slightly imperfectly.

I suspect that modern computers could do dynamic tuning but doubt that this would be "perfect" either. If you tune one chord to perfection and play it on an organ then its reverberation will still be audible when the next slightly differently tuned but perfect chord is played - the two will then clash and become audibly imperfect again. I also doubt if an algorithm would be able to predict the future.

I liked the referenceed pdf but it is a very hard read and note that the potentially interesting internally referenced web page is no longer on line.

csw900
